workflow/controller/steps.go

```diff
@@ -494,6 +495,26 @@ func (woc *wfOperationCtx) prepareMetricScope(node *wfv1.NodeStatus) (map[string
 	realTimeScope := make(map[string]func() float64)
 	localScope := woc.globalParams.DeepCopy()
 
+	if node == nil {
```

Comment:
So do this instead?
```
if node == nil {
    return prepareDefaultMetricScope(...)
}
```

Reply:
I was thinking more like

```go
	if woc.execWf.Spec.Metrics != nil {
		localScope, realTimeScope := woc.prepareDefaultMetricScope()
		woc.computeMetrics(woc.execWf.Spec.Metrics.Prometheus, localScope, realTimeScope, true)
	}
```

I am not comfortable with `prepareMetricScope` accepting `nil` as a valid parameter

Reply:
Or perhaps this is better:

```go
	if woc.execWf.Spec.Metrics != nil {
		realTimeScope := map[string]func() float64{common.GlobalVarWorkflowDuration: func() float64 {
			return time.Since(woc.wf.Status.StartedAt.Time).Seconds()
		}}
		localScope := woc.addDefaultMetricsTo(woc.globalParams)
		woc.computeMetrics(woc.execWf.Spec.Metrics.Prometheus, localScope, realTimeScope, true)
	}
```

Reply:
Since `prepareMetricScope` is used elsewhere I would prefer not to change it

Reply:
Added default metric scope function in https://github.com/argoproj/argo-workflows/pull/7299/commits/ec2d843ce0ad0b33dd34b8d9320c6121d41f7453.